source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -70,6 +70,9 @@ DnsCacheImpl::DnsCacheImpl(
   }
   enable_dfp_dns_trace_ = context.serverFactoryContext().runtime().snapshot().getBoolean(
       "envoy.enable_dfp_dns_trace", false);
+  allow_disabling_dns_query_timeout_ =
+      context.serverFactoryContext().runtime().snapshot().getBoolean(
+          "envoy.allow_disabling_dns_query_timeout", true);
```

Comment:
Also as the DNS cache can change on cluster / route change this may not truly be restart , it's kind of a long persisting reloadable :-/